TSPL.docc/ReferenceManual/Expressions.md

```diff
@@ -921,6 +921,93 @@ to make use of the implementation in their superclass.
 >
 > *superclass-initializer-expression* → **`super`** **`.`** **`init`**
 
+### Conditional Expression
+
+A *conditional expression* evaluates to one of several given values
+based on the value of a condition.
+It has one the following forms:
+
+```swift
+if <#condition 1#> {
+   <#expression used if condition 1 is true#>
+} else if <#condition 2#> {
+   <#expression used if condition 2 is true#>
+} else {
+   <#expression used if both conditions are false#>
+}
+
+switch <#expression#> {
+case <#pattern 1#>:
+    <#expression 1#>
+case <#pattern 2#> where <#condition#>:
+    <#expression 2#>
+default:
+    <#expression 3#>
+}
+```
+
+A conditional expression
+has the same behavior and syntax as an `if` statement or a `switch` statement,
+except for the differences listed below.
+
+A conditional expression appears only in the following contexts:
+
+  - As the value assigned to a variable.
+  - As the initial value in a variable or constant declaration.
+  - As the error thrown by a `throw` expression.
+  - As the value returned by a function, closure, or property getter.
+
+The branches of a conditional expression are exhaustive,
+ensuring that the expression always produces a value
+regardless of the condition.
+This means each `if` branch needs a corresponding `else` branch.
+
+Each branch contains either a single expression,
+which is used as the value for the conditional expression
+when that branch's conditional is true,
+a `throw` statement,
+or a call to a function that never returns.
+
+Each branch must produce a value of the same type.
+Because type checking of each branch is independent,
+conditional expressions that that include a `nil` literal
+or different kinds of literals typically need type context,
+like an explicit type annotation.
+
+```swift
+let number: Double = if someCondition { 10 } else {12.34 }
+```
+
+A conditional expression can't appear inside a result builder.
```

Comment:
The SE proposal says:

> The expression is not part of a result builder expression

But also says:

> The variable declaration form of an `if` will be allowed in result builders.

Which seems contradictory.  What did we mean?

It sounds like the intention might be that an `if` or `switch` expression doesn't get transformed when it's in a result builder — that is, there's no call to `buildEither(first:)` or `buildEither(second:)` — but that you can use `if` and `switch` expressions to set the value of a variable inside of a result builder.

Reply:
Thanks! Rewrote in faec17e0.